### PR TITLE
feat(migration-tool): scrape bodiless data

### LINKED ISSUE #645: Scrape to bodiless data
## User Story
As a site builder
I want to scrape various page-level data from an existing site
So that content entry is quicker

## Acceptance Criteria

AC1. It is possible to configure the migration tool to scrape data from site pages and write it to json files in the directory of the corresponding page, or in the side data directory.

AC2. Configuration of data to scrape may specified either as a list of JS modules or (if configuration files are migrated from .json to.js) as a list of functions.  In the first case, he default export of each modules is a function. In either case, the functionreceives as arguments:
- the parsed HTML for the page
- the page path
- a MigrationApi object which contains methods to write files to the page or site directory.
- an object containing additional parameters specified as part of the configuration.

eg (in migrationconfig.json)
```
{
   ...,
   "dataScrapers": {
      "foo.js",
      {
        "resolve": "bar.js",
        "params": {
          "baz": "bang"
        }
     },
   ...
```
or, in migration-config.js
```
dataScrapers: [
  foo,
  bar({ baz: bang }),
],
```

AC3. The migration tool includesa a predefined scraper for metatags.  This acceps a list of names and a prefix.  For each name in the list it creates a json file at`${prefix}/${name}.json` with the content
```
{
  content: `${value}`
}
```

AC4. There is a reusable migration tool configuration for canvas migration which includes a data scraper (converter) which writes an index.json file using data from the page (eg body class, page-type metatag and/or page path) to set templates correctly for 
- product listing page
- product detail page

scrapeTemplate.js

```
export default ({ document, path, api }) => {
  const pageType = doc.find('meta[name="page-type"]').attr('value');
  const bodyClass = doc.find('body').attr('class').match(/node-type-(.*) /)?[0];
  if (pageType === 'product-listing') {
     api.writeJsonSync('index.json', { '#template': 'product_listing', '#child-template': 'product_detail' });
  } else if (pageType === 'product' || bodyClass === 'product' || pagePath.match(/^\/products)) {
     api.writeJsonSync('index.json', { '#template': 'product_detail' });
  } else if( ... ) {
    ...
  } else {
     api.writeJsonSync('index.json', { '#template': '_default' });
  }
```

  
   ...
```

### PR DESCRIPTION
**Pull request by beliayeu**
## Changes
* provided api to scrape data from site pages and write it to json files in the directory of the corresponding page, or in the side data directory
* included a predefined scraper for metatags that implements api mentioned above.

## Related Issues

Fixes #645

### CODE REVIEW

File: packages/bodiless-migration-tool/examples/settings/gdpr_prod_to_test.js.sample [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ module.exports = {
+  /**
+   * page url that should be scraped
+   */
+  url: "https://www.imodium.de/",
```
Comment (dewen) [inline packages/bodiless-migration-tool/examples/settings/gdpr_prod_to_test.js.sample:19]: Shall we avoid real domain in sample?

File: packages/bodiless-migration-tool/examples/settings/trailing_slash_remove.js.sample [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ module.exports = {
+  /**
+   * page url that should be scraped
+   */
+  url: "https://pariet10.ru/",
```
Comment (dewen) [inline packages/bodiless-migration-tool/examples/settings/trailing_slash_remove.js.sample:19]: Shall we avoid real domain in sample?

File: packages/bodiless-migration-tool/src/helpers.ts [hunk 3]
```diff
@@ -185,3 +185,16 @@ export function prependProtocolToBareUrl(url$: string, protocol = 'https://') {
...
   return url$;
 }
+
+/**
+ * function that strips extension from url
+ * @param pageUrl - page url from which extension should be stripped
+ * @returns url - path url with stripped extension
+ *
+ * @example
+ * // return /products
```
Comment (dewen) [inline packages/bodiless-migration-tool/src/helpers.ts:195]: extra "//"?

File: packages/bodiless-migration-tool/examples/settings/replace_string_in_html_source.js.sample [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ module.exports = {
+  /**
+   * page url that should be scraped
+   */
+  url: "https://lepetitmarseillais.be/",
```
Comment (dewen) [inline packages/bodiless-migration-tool/examples/settings/replace_string_in_html_source.js.sample:19]: Shall we avoid real domain in sample?

File: packages/bodiless-migration-tool/examples/settings/trailing_slash_add.js.sample [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ module.exports = {
+  /**
+   * page url that should be scraped
+   */
+  url: "https://pariet10.ru/",
```
Comment (dewen) [inline packages/bodiless-migration-tool/examples/settings/trailing_slash_add.js.sample:19]: Shall we avoid real domain in sample?

**Review by dewen — APPROVED** [review]

**Comment by marcopagliarulo** [comment]
Hi @beliayeu 
The test of metatags plugin is failed because there isn't a way to provide a list of names as defined by the AC3

**Review by wodenx — APPROVED** [review]

**Review by marcopagliarulo — APPROVED** [review]

File: packages/bodiless-migration-tool/examples/settings/trailing_slash_remove.js.sample (hunk 2 above)
Comment (beliayeu) [reply packages/bodiless-migration-tool/examples/settings/trailing_slash_remove.js.sample:19]: per discussion in skype that is not a concern

**Pull request merged**